src/test/java/itest/GraphQLIT.java

```diff
@@ -588,6 +602,475 @@ void testNodesHaveIds() throws Exception {
+
+        // verify recording created
+        CompletableFuture<JsonArray> listRespFuture2 = new CompletableFuture<>();
+        webClient
+                .get(String.format("/api/v1/targets/%s/recordings", SELF_REFERENCE_TARGET_ID))
+                .send(
+                        ar -> {
+                            if (assertRequestStatus(ar, listRespFuture2)) {
+                                listRespFuture2.complete(ar.result().bodyAsJsonArray());
+                            }
+                        });
+        listResp = listRespFuture2.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+        MatcherAssert.assertThat(
+                "list should have size 1 after recording creation",
+                listResp.size(),
+                Matchers.equalTo(1));
+        JsonObject recordingInfo = listResp.getJsonObject(0);
+        MatcherAssert.assertThat(
+                recordingInfo.getString("name"), Matchers.equalTo(TEST_RECORDING_NAME));
+        MatcherAssert.assertThat(recordingInfo.getString("state"), Matchers.equalTo("RUNNING"));
+
+        Thread.sleep(2_000L);
+        for (int i = 0; i < 2; i++) {
+            // Archive the recording
+            CompletableFuture<Void> archiveRecordingFuture = new CompletableFuture<>();
+            webClient
+                    .patch(
+                            String.format(
+                                    "/api/v1/targets/%s/recordings/%s",
+                                    SELF_REFERENCE_TARGET_ID, TEST_RECORDING_NAME))
+                    .putHeader(HttpHeaders.CONTENT_TYPE.toString(), HttpMimeType.PLAINTEXT.mime())
+                    .sendBuffer(
+                            Buffer.buffer("SAVE"),
+                            ar -> {
+                                if (assertRequestStatus(ar, archiveRecordingFuture)) {
+                                    archiveRecordingFuture.complete(null);
+                                } else {
+
+                                    archiveRecordingFuture.completeExceptionally(
+                                            new RuntimeException("Archive request failed"));
+                                }
+                            });
+
+            archiveRecordingFuture.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+        }
+
+        // GraphQL Query to filter Archived recordings by names
+        CompletableFuture<TargetNodesQueryResponse> resp2 = new CompletableFuture<>();
+        // List<String> filterNames = Arrays.asList("Recording1", "Recording2");
+        String query =
+                "query { targetNodes {"
+                        + "recordings {"
+                        + "archived(filter: { names: [\"io-cryostat-Cryostat_"
+                        + TEST_RECORDING_NAME
+                        + "_\\\\d{8}T\\\\d{6}Z(\\\\.\\\\d+)?\\\\.jfr\",\"io-cryostat-Cryostat_"
+                        + TEST_RECORDING_NAME
+                        + "_\\\\d{8}T\\\\d{6}Z(\\\\.\\\\d+)?\\\\.jfr\"] }) {"
```

Comment:
How does this work? I thought the implementation was looking for exact string matches for the recording names, but this test is passing a regular expression pattern as the recording name and expecting the backend to perform a regex match during the query. Where is this regular expression getting compiled and matched against... ?